### PR TITLE
Add policy event triggers for Hawkular

### PR DESCRIPTION
**Pull request by tumido**
Create mapping for Hawkular events so they trigger Middleware events via automate. 

Depends on: https://github.com/ManageIQ/manageiq/pull/16134
Required by: https://github.com/ManageIQ/manageiq-providers-hawkular/pull/71
Resolves: https://issues.jboss.org/browse/JMAN4-200

### CODE REVIEW

**Comment by tumido** [comment]
cc @israel-hdez, so after all ([comment](https://github.com/ManageIQ/manageiq-providers-hawkular/pull/71#issuecomment-334815536)) the automate is actually needed to trigger the policy events... ;)

cc @lfu is this how it should look like?
cc @abonas, this is gonna solve the issue with events I ran into, we're moving!

**Comment by lfu** [comment]
@tumido Forgot to mention that we need to follow the steps in [Creating a Pull Request for enhancements to ManageIQ Automate Domain](http://talk.manageiq.org/t/creating-a-pull-request-for-enhancements-to-manageiq-automate-domain/1041) to keep the YAML files format consistent.

File: content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_datasource.error.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  attributes:
+    display_name: Datasource Error
+    name: hawkular_datasource.error
+    inherits:
+    description:
+  fields:
+  - rel4:
+      value: "/System/event_handlers/event_action_refresh?target=ems"
+  - rel5:
+      value: "/System/event_handlers/event_action_policy?target=middleware_server&policy_event=mw_datasource_error&param="
```
Comment (lfu) [inline content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_datasource.error.yaml]: @tumido ```policy_event``` raised in this PR does not match what is defined in [your PR for MiqEventDefinition](https://github.com/ManageIQ/manageiq/pull/16134/files#diff-da17f6bf8d1c54c7878221e8110a0577R77).

**Comment by tumido** [comment]
@lfu, it should be fixed now ;)

**Comment by abonas** [comment]
@miq-bot add_label providers/middleware

**Comment by abonas** [comment]
1. what are the sections rel4 and rel5 for? 
2. what is the functionality that is enabled by these files? it seems a lot of overhead to create and maintain  those files per every single event. (or I am missing something?)
(the above are mainly questions to core team)
cc @agrare

**Comment by gmcculloug** [comment]
These files are defining the actions taken when an event is sent into automate.  rel4 is sending a refresh to the provider and rel5 is raising the corresponding policy event for the provider event that the file is handling.

The files are part of the export format for the automate model.

**Comment by abonas** [comment]
@gmcculloug does this mean that it *always* sends a request to refresh when this event arrives?
@tumido have you considered if rel4 is necessary in all cases?

**Comment by gmcculloug** [comment]
Yes, `event_action_refresh?target=ems` is requesting a full provider refresh each time the event is processed.

**Comment by tumido** [comment]
@abonas, yes I think the refresh is necessary, but I would like to hear @israel-hdez opinion on that matter. But I generally think, since the event is notifying about some action on a MiddlewareServer, it's worth refreshing to see update inventory if for example the removal of deployment failed etc... But I'm not 100% sure about that. So if you consider the refresh is not necessary in some cases, I don't mind to remove it.

The "rel4", "rel5" are just slots taken by the action. There are multiple slots available and they are executed in the name order. So if you have defined "rel1", "rel2" and "rel4" they are execuded in order 1,2,4, etc. User can modify the mapping in ManageIQ and put custom actions in the empty slots. so using "rel4" and rel5" allows user to put his custom actions before and after our actions.

**Comment by abonas** [comment]
@tumido my concern is that refresh is a heavy action. and might not be required for all the cases. hence it will be good to review each case separately in that context.

**Comment by tumido** [comment]
@abonas, yes I agree. @israel-hdez has much better grip on what's going on under the hood when such event happens, so he know better. I don't feel educated enough to decide. To me it seems doing a refresh is safer and keeps us consistent if we're not sure...

**Comment by israel-hdez** [comment]
@tumido I would skip the refresh for the error events. I think it's not needed, since we know nothing was added/removed from inventory.
Also, the JDBC drivers aren't inventoried; they are fetched directly from Hawkular when required. The refresh for those cases surely is not needed.

**Comment by tumido** [comment]
@israel-hdez done, please review

File: content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_datasource.ok.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+object_type: instance
+version: 1.0
+object:
+  attributes:
+    display_name: Datasource Success
```
Comment (abonas) [inline content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_datasource.ok.yaml]: what Datasource Success means? what kind of event is this? I think I'm missing a verb here :)

File: content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_jdbc.ok.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+object_type: instance
+version: 1.0
+object:
+  attributes:
+    display_name: JDBC Success
```
Comment (abonas) [inline content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_jdbc.ok.yaml]: missing a verb/action here. what does it mean JDBC success?

Comment (tumido) [reply content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_jdbc.ok.yaml]: I've tried to follow what we already have [here](https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_deployment.ok.yaml) (I considered Deployment as a noun not a verb, my bad). I'm gonna change it so we're more consistent with the https://github.com/ManageIQ/manageiq/pull/16134.

**Comment by tumido** [comment]
@abonas, is this better?

File: content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_deployment_remove.ok.yaml [hunk 4]
```diff
@@ -3,7 +3,7 @@ object_type: instance
 version: 1.0
 object:
   attributes:
-    display_name: Deployment Remove Success
+    display_name: Application Uneploy Success
```
Comment (abonas) [inline content/automate/ManageIQ/System/Event/EmsEvent/Hawkular.class/hawkular_deployment_remove.ok.yaml]: typo, missing "d" in undeploy

**Pull request merged**